File: tests/fixtures/codegen/inputs/pointers-expected/queries/items/insertpayload.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
)

const InsertPayloadSQL = "INSERT INTO items (payload) VALUES ($1);"

type InsertPayloadParams struct {
	Payload jsonschemas.Event `db:"payload"`
}

func (q *Queries) InsertPayload(ctx context.Context, params InsertPayloadParams) error {
	jsonInput1, err := pgsidpgx.ValidateJSONInput(params.Payload, "pgsid:///jsonschemas/Event.json#", "InsertPayload", "payload", false)
	if err != nil {
		return err
	}
	_, err = q.db.Exec(ctx, InsertPayloadSQL, jsonInput1)
	return err
}

File: tests/fixtures/codegen/inputs/pointers-expected/typescript/runtime/queries/items/InsertPayload.ts
import type { InsertPayloadParams } from "../../../types/queries/items/InsertPayload.js";
export type { InsertPayloadParams } from "../../../types/queries/items/InsertPayload.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const insertPayloadSql = "INSERT INTO items (payload) VALUES ($1);";
export function isInsertPayloadParamsPayloadPublicItemsPayload(value: unknown): value is InsertPayloadParams["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateInsertPayloadParamsPayloadPublicItemsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export async function insertPayload(db: pgsid.Queryable, params: InsertPayloadParams): Promise<void> {
    const jsonInput1 = JSON.stringify(params["payload"]);
    if (jsonInput1 === undefined)
        throw new pgsid.QueryValidationError("InsertPayload", "payload", [{ path: [], keyword: "json", expected: "JSON value", received: "undefined", message: "Value has no JSON representation" }]);
    const jsonInput1Value = jsonInput1 === null ? null : JSON.parse(jsonInput1);
    if (jsonInput1 !== null) {
        if (!isInsertPayloadParamsPayloadPublicItemsPayload(jsonInput1Value))
            throw new pgsid.QueryValidationError("InsertPayload", "payload", validateInsertPayloadParamsPayloadPublicItemsPayload(jsonInput1Value).issues);
    }
    await db.query(insertPayloadSql, [jsonInput1]);
}
